fastn-core/src/migrations.rs

```diff
@@ -0,0 +1,210 @@
+const MIGRATION_TABLE: &str = r#"
+
+CREATE TABLE IF NOT EXISTS
+    fastn_migration
+(
+    id               INTEGER PRIMARY KEY,
+    app_name         TEXT NOT NULL,
+    migration_number INTEGER NOT NULL UNIQUE,
+    migration_name   TEXT NOT NULL,
+    applied_on       INTEGER NOT NULL
+) STRICT;
+
+"#;
+
+pub(crate) async fn migrate(config: &fastn_core::Config) -> Result<(), MigrationError> {
+    let db = config.get_db_url().await;
+    create_migration_table(config, db.as_str()).await?;
```

Comment:
In general if you are passing `x`, and a data derived from `x`, the consider just passing `x`.

Reply:
Make `migrate()` a method of `fastn_core::Config`. Same as `has_new_migration`.

Reply:
Leave the implementations in this file only. 

```rs
impl Config {
    pub(crate) fn migrate() {}
    pub(crate) fn has_unapplied_migrations() {}
}
```

Reply:
Not following this in favor of https://github.com/fastn-stack/fastn/pull/1894#discussion_r1627130340